packages/turf-intersect/index.js

```diff
@@ -34,22 +35,17 @@ var jsts = require('jsts');
  * var addToMap = [poly1, poly2, intersection];
  */
 module.exports = function (poly1, poly2) {
-    var geom1, geom2;
-    if (poly1.type === 'Feature') geom1 = poly1.geometry;
-    else geom1 = poly1;
-    if (poly2.type === 'Feature') geom2 = poly2.geometry;
-    else geom2 = poly2;
+    var geom1 = (poly1.type === 'Feature') ? poly1.geometry : poly1;
+    var geom2 = (poly2.type === 'Feature') ? poly2.geometry : poly2;
+
     var reader = new jsts.io.GeoJSONReader();
-    var a = reader.read(JSON.stringify(geom1));
-    var b = reader.read(JSON.stringify(geom2));
+    var a = reader.read(JSON.stringify(truncate(geom1)));
+    var b = reader.read(JSON.stringify(truncate(geom2)));
     var intersection = a.intersection(b);
 
-    if (intersection.isEmpty()) {
-        return undefined;
-    }
+    if (intersection.isEmpty()) return null;
```

Comment:
No worries 👍 

Well I think we are all in agreement that if you have no results in collection of Features, then we would have an empty FeatureCollection, if the output is a single Feature which would be `undefined` or `null` then we would output a `Feature<null>` that way things don't blow up when you chain TurfJS methods by sending a `null` as input to another module.